packages/yup/README.md

```diff
@@ -0,0 +1,98 @@
+# yup
+> Method extensions for the [yup](https://github.com/jquense/yup)
+
+[![Version](https://img.shields.io/npm/v/@availity/yup.svg?style=for-the-badge)](https://www.npmjs.com/package/@availity/yup)
+
+## Install
+
+```bash
+npm install @availity/yup yup --save
+```
+
+## Usage
+
+Import the package in the root of your project somewhere and you will have access to all of the provided yup methods.
+
+```javascript
+import '@availity/yup'
+import * as yup from 'yup';
+
+const schema = yup.string().dateFormat('MM/DD/YYYY');
+
+schema.isValid('12-12-2012');
+```
+
+## Table of Contents
+
+- [isRequired](#isRequired)
+- [dateRange](#dateRange)
+- [dateFormat](#dateFormat)
+- [between](#between)
+
+> Note any date validation methods will required [moment](https://www.npmjs.com/package/moment) to be installed
+
+
+
+## Methods
+### isRequired [**String**,**Array**,**Number**]
+#### Parameters
+- **required** - `boolean`. **required**. Whether or not the given string is required.
+- **message** - `string`. Optional. Custom error message when invalid. Default: `This Field is Required.`
+
+#### Example
+```javascript
+yup.string().isRequired();
+yup.string().isRequired(true,"Custom Error Message");
+yup.number().isRequired();
+yup.array().isRequired();
+```
+
+### dateRange [**Object**]
+Evaluates a date range object.
+#### Parameters
+- **options** - `object`. **required**. Range Options.
+  - **min** - `string`. **required**. The min date that can be selected.
+  - **max** - `string`. **required**. The max date that can be selected.
+  - **format** - `string`. optional. The format to parse the dates with.
```

Comment:
What is the default format?

Reply:
Yeah I have the default as `MM/DD/YYYY`. Will update it. 

I was sort of trying to mirror the same validation logic we had in `reactstrap-validation-date`. Not sure if that is something you would recommend or anything you would have done differently with the validation logic in the date library or the standard form library.

[reactstrap-validation-date formats](https://github.com/Availity/availity-react/blob/master/packages/reactstrap-validation-date/AvDate.js#L108)
[reactstrap-validation dateRange](https://github.com/Availity/availity-reactstrap-validation/blob/master/src/AvValidator/dateRange.js)

Reply:
for type="date" the format will be ISO (`YYYY-MM-DD`) for browsers which support it or the locale format for browsers which do not (or well any format for the browsers which do not support it since it doesn't modify the input).
I try to account for that plus, if people wanted to enter `11102012` or `20121110` (with or without separating character) it would accept it and autoformat it. It was mostly for people who paste values and those values could be in various other formats (client feedback suggested that we don't force the users to enter `/` and we allow pasting of various formats).
Here is just validation, so I am not sure if it is desired to allow the autoformating since it cannot manipulate the value; only validate it.

Reply:
Ended up just allowing one support format when validating the field.